internal/tlscert/tls_cert_validator.go

```diff
@@ -96,34 +130,64 @@ func (v *Validator) ValidateCertificate(ctx context.Context, cert, key *telemetr
 		return err
 	}
 
+	// Parse the CA
+	parsedCAs, err := parseCertificates(sanitizedCA, ErrCADecodeFailed, ErrCAParseFailed)
+	if err != nil {
+		return err
+	}
+
+	// Validate the certificate-key pair
 	_, err = tls.X509KeyPair(sanitizedCert, sanitizedKey)
 	if err != nil {
 		return ErrInvalidCertificateKeyPair
 	}
 
-	if v.now().After(certExpiry) {
-		return &CertExpiredError{Expiry: certExpiry}
+	// Validate CA(s)
+	now := v.now()
+	for _, ca := range parsedCAs {
+		if err = validateCA(ca, now); err != nil {
+			return err
+		}
 	}
 
-	if certExpiry.Sub(v.now()) <= twoWeeks {
-		return &CertAboutToExpireError{Expiry: certExpiry}
+	// Validate certificate expiry
```

Comment:
Let's move this for consistency also to a separate `validateCertificate` method.

Reply:
Maybe including the cert / key matching check above